src/dialog/unix/SDL_zenitydialog.c

```diff
@@ -225,29 +145,33 @@ static char** generate_args(const zenityArgs* info)
                                               "*.", " *.", "");
 
             if (!filter_str) {
-                CLEAR_AND_RETURN()
+                while (i--) {
+                    SDL_free((void *)argv[--nextarg]);
```

Comment:
You shouldn't free all those `const char *` that you assigned above, or am I missing something?